### ISSUE TITLE
Test_openjdk8_j9_sanity.system_x86-64_linux MathLoadTest_bigdecimal_0 crash libj9vm29.so vmState=0x00000000

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#6474]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_sanity.system_x86-64_linux_Nightly/97
```
MLT 06:03:33.136 - Starting thread. Suite=0 thread=6
MLT stderr Unhandled exception
MLT stderr Type=Segmentation error vmState=0x00000000
MLT stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000002
MLT stderr Handler1=00007F5AF77DF1F0 Handler2=00007F5AF70D9520 InaccessibleAddress=00007F5ADF2C3C1B
MLT stderr RDI=0000000000BE7E00 RSI=00000000FB7CF200 RAX=00000000E018C4D0 RBX=0000000000000000
MLT stderr RCX=00000000E0465A98 RDX=0000000000000001 R8=0000000000BE7E00 R9=0000000000000001
MLT stderr R10=0000000000ADD600 R11=00000000FB7CF1E8 R12=0000000000000006 R13=000000000000000A
MLT stderr R14=00000000FB7CF1E8 R15=00000000E0186BA0
MLT stderr RIP=00007F5ADF2C3C1B GS=0000 FS=0000 RSP=0000000000C3B878
MLT stderr EFlags=0000000000010246 CS=0033 RBP=00000000009B3100 ERR=0000000000000015
MLT stderr TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=00007F5ADF2C3C1B
MLT stderr xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MLT stderr xmm1 6f0adca4e4f10983 (f: 3841001984.000000, d: 7.954325e+226)
MLT stderr xmm2 00007f5aa4c307f0 (f: 2764244992.000000, d: 6.918267e-310)
MLT stderr xmm3 ffff000000000000 (f: 0.000000, d: -nan)
MLT stderr xmm4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MLT stderr xmm5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MLT stderr xmm6 00000000ff000000 (f: 4278190080.000000, d: 2.113707e-314)
MLT stderr xmm7 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
MLT stderr xmm8 4010000000000000 (f: 0.000000, d: 4.000000e+00)
MLT stderr xmm9 400aaaaaaaaaaaab (f: 2863311616.000000, d: 3.333333e+00)
MLT stderr xmm10 3d7924171d86722d (f: 495350304.000000, d: 1.429099e-12)
MLT stderr xmm11 4031fe6078cba891 (f: 2026612864.000000, d: 1.799366e+01)
MLT stderr xmm12 bcba000000000000 (f: 0.000000, d: -3.608225e-16)
MLT stderr xmm13 3f9cd7d97e93bc00 (f: 2123611136.000000, d: 2.816715e-02)
MLT stderr xmm14 403205966f2b4d80 (f: 1865108864.000000, d: 1.802183e+01)
MLT stderr xmm15 3c3d5746e55acbc0 (f: 3847932928.000000, d: 1.590575e-18)
MLT stderr Target=2_90_20190714_95 (Linux 4.4.0-140-generic)
MLT stderr CPU=amd64 (8 logical CPUs) (0x3eae99000 RAM)
MLT stderr ----------- Stack Backtrace -----------
MLT stderr (0x00007F5AF70F59D2 [libj9prt29.so+0x3a9d2])
MLT stderr (0x00007F5AF70DA283 [libj9prt29.so+0x1f283])
MLT stderr (0x00007F5AF70F5A4E [libj9prt29.so+0x3aa4e])
MLT stderr (0x00007F5AF70F5B44 [libj9prt29.so+0x3ab44])
MLT stderr (0x00007F5AF70DA283 [libj9prt29.so+0x1f283])
MLT stderr (0x00007F5AF70F58AB [libj9prt29.so+0x3a8ab])
MLT stderr (0x00007F5AF77DED74 [libj9vm29.so+0x9ed74])
MLT stderr (0x00007F5AF70DA283 [libj9prt29.so+0x1f283])
MLT stderr (0x00007F5AF77DEFC8 [libj9vm29.so+0x9efc8])
MLT stderr (0x00007F5AF780B485 [libj9vm29.so+0xcb485])
MLT stderr ---------------------------------------
```

**Comment by gacholio** [comment]
Looking at the contents of RSP, it appears there's a JIT return address there:

(gdb) x/20gx 0x0000000000C3B878
0xc3b878:	0x00007f5ae029b281	0x00000000e05ea1b8
(gdb) x/20i 0x00007f5ae029b281-5
   0x7f5ae029b27c:	callq  0x7f5ae02c3c1b
   0x7f5ae029b281:	jmp    0x7f5ae029b289

> !jitmetadatafrompc 0x7f5ae029b27c
J9JITExceptionTable at 0x00007F5ADCBE2578 {
    struct J9UTF8* className = !j9utf8 0x00007F5AA5670B80   // java/util/regex/Pattern$Curly
    struct J9UTF8* methodName = !j9utf8 0x00007F5AA56730B8   // match0
    struct J9UTF8* methodSignature = !j9utf8 0x00007F5AA56730C0   // (Ljava/util/regex/Matcher;IILjava/lang/CharSequence;)Z
    struct J9ConstantPool* constantPool = !j9constantpool 0x0000000000AD43D0 
    struct J9Method* ramMethod = !j9method 0x0000000000AD45A8   // java/util/regex/Pattern$Curly.match0(Ljava/util/regex/Matcher;IILjava/lang/CharSequence;)Z
    UDATA parm.startPC = 0x00007F5AE029B208;
    UDATA parm.endWarmPC = 0x00007F5AE029B54F;
    UDATA parm.startColdPC = 0x0000000000000000;
    UDATA parm.endPC = 0x00007F5AE029B54F;
    UDATA parm.totalFrameSize = 0x000000000000000B;
    I_16 parm.slots = 0x0005;
    I_16 parm.scalarTempSlots = 0x0000;
    I_16 parm.objectTempSlots = 0x0003;
    U_16 parm.prologuePushes = 0x0002;
    I_16 parm.tempOffset = 0x0000;
    U_16 parm.numExcptionRanges = 0x0000;
    I_32 parm.size = 0x000001E0;
    UDATA parm.registerSaveDescription = 0x0000000000000202;
    void* gcStackAtlas = !void 0x00007F5ADCBE2650 
    void* inlinedCalls = !void 0x0000000000000000 
    void* bodyInfo = !void 0x00007F5A7C96EF70 
    struct J9JITExceptionTable* nextMethod = !j9jitexceptiontable 0x00007F5ADCBE2338 
    struct J9JITExceptionTable* prevMethod = !j9jitexceptiontable 0x00007F5ADCBE2878 
    void* debugSlot1 = !void 0x0000000000000000 
    void* debugSlot2 = !void 0x0000000000000000 
    void* osrInfo = !void 0x0000000000000000 
    void* runtimeAssumptionList = !void 0x00007F5A7C96EF40 
    I_32 parm.hotness = 0x00000002;
    UDATA parm.codeCacheAlloc = 0x00007F5AE029B1D0;
    void* gpuCode = !void 0x0000000000000000 
    void* riData = !void 0x0000000000000000 
}

Called method is:

> !jitmetadatafrompc 0x7f5ae02c3c1b
J9JITExceptionTable at 0x00007F5ADCBA54F8 {
    struct J9UTF8* className = !j9utf8 0x00007F5AA5671810   // java/util/regex/Pattern$BmpCharProperty
    struct J9UTF8* methodName = !j9utf8 0x00007F5AA553578A   // match
    struct J9UTF8* methodSignature = !j9utf8 0x00007F5AA5670888   // (Ljava/util/regex/Matcher;ILjava/lang/CharSequence;)Z
    struct J9ConstantPool* constantPool = !j9constantpool 0x0000000000AD3CD0 
    struct J9Method* ramMethod = !j9method 0x0000000000AD3A00   // java/util/regex/Pattern$BmpCharProperty.match(Ljava/util/regex/Matcher;ILjava/lang/CharSequence;)Z
    UDATA parm.startPC = 0x00007F5AE02C3C08;
    UDATA parm.endWarmPC = 0x00007F5AE02C3EA5;
    UDATA parm.startColdPC = 0x0000000000000000;
    UDATA parm.endPC = 0x00007F5AE02C3EA5;
    UDATA parm.totalFrameSize = 0x0000000000000009;
    I_16 parm.slots = 0x0004;
    I_16 parm.scalarTempSlots = 0x0000;
    I_16 parm.objectTempSlots = 0x0003;
    U_16 parm.prologuePushes = 0x0002;
    I_16 parm.tempOffset = 0x0000;
    U_16 parm.numExcptionRanges = 0x0000;
    I_32 parm.size = 0x000001CE;
    UDATA parm.registerSaveDescription = 0x0000000000000202;
    void* gcStackAtlas = !void 0x00007F5ADCBA5603 
    void* inlinedCalls = !void 0x00007F5ADCBA55D0 
    void* bodyInfo = !void 0x00007F5A7C7F5090 
    struct J9JITExceptionTable* nextMethod = !j9jitexceptiontable 0x00007F5ADCBB8CF8 
    struct J9JITExceptionTable* prevMethod = !j9jitexceptiontable 0x00007F5ADCBFA2F8 
    void* debugSlot1 = !void 0x0000000000000000 
    void* debugSlot2 = !void 0x0000000000000000 
    void* osrInfo = !void 0x0000000000000000 
    void* runtimeAssumptionList = !void 0x00007F5A7C7F5060 
    I_32 parm.hotness = 0x00000002;
    UDATA parm.codeCacheAlloc = 0x00007F5AE02C3BD0;
    void* gpuCode = !void 0x0000000000000000 
    void* riData = !void 0x0000000000000000 
}

**Comment by gacholio** [comment]
The crashing address is suspiciously similar to the called address:

0x00007F5ADF2C3C1B (crash)
0x00007F5AE02C3C1B (call)

Wondering if this is a patching issue (the call takes place before the instruction has been fully patched).

**Comment by gacholio** [comment]
0x7f5ae029b27c:	0xe8	0x9a	0x89	0x02	0x00

That's the call instruction.

**Comment by gacholio** [comment]
@andrewcraik Not sure who works on X these days.

**Comment by pshipton** [comment]
Haven't seen this again.

**Comment by roded** [comment]
We've experiencing what seems to be a similar issue when running Tomcat on OpenJ9 `jdk8u222-b10_openj9-0.15.1`  - see exception below. Can anyone confirm that this issue is in fact the same and if it was resolved in a later version of OpenJ9?

`Unhandled exception
Type=Segmentation error vmState=0x00000000
J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
Handler1=00007F9AA2193000 Handler2=00007F9AA1A8D520 InaccessibleAddress=0000000000000000
RDI=00007F9A740FA440 RSI=00000007FD4368BD RAX=00007F9A740C6D60 RBX=0000000000000050
RCX=0000000000000050 RDX=00000007FD436AA0 R8=0000000000000050 R9=0000000000000000
R10=0000000001D3C598 R11=00007F9A4A783210 R12=00000007FD436AA0 R13=00000007FD4368BD
R14=00007F9A5C0677DC R15=0000000000000000
RIP=0000000000000000 GS=0000 FS=0000 RSP=00007F9A5C067708
EFlags=0000000000010246 CS=0033 RBP=00007F9A740FA440 ERR=0000000000000014
TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=0000000000000000
xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm1 43e0000000000000 (f: 0.000000, d: 9.223372e+18)
xmm2 00007f9a5c0677f0 (f: 1543927808.000000, d: 6.931788e-310)
xmm3 00000000400ab7a6 (f: 1074444160.000000, d: 5.308460e-315)
xmm4 000000004a48d0c4 (f: 1246286080.000000, d: 6.157471e-315)
xmm5 3bbcc86800000000 (f: 0.000000, d: 6.095003e-21)
xmm6 40314025e917b690 (f: 3910645504.000000, d: 1.725058e+01)
xmm7 40314025e917b690 (f: 3910645504.000000, d: 1.725058e+01)
xmm8 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
xmm9 40314026598a543c (f: 1502237696.000000, d: 1.725059e+01)
xmm10 3ff14cb23a997398 (f: 983135104.000000, d: 1.081225e+00)
xmm11 3ccd000000000000 (f: 0.000000, d: 8.049117e-16)
xmm12 bc42000000000000 (f: 0.000000, d: -1.951564e-18)
xmm13 4031542457337bc0 (f: 1462991872.000000, d: 1.732868e+01)
xmm14 bbe9abe6bf0fa436 (f: 3205473280.000000, d: -4.348920e-20)
xmm15 4031542457337bc0 (f: 1462991872.000000, d: 1.732868e+01)
Target=2_90_20190717_368 (Linux 3.10.0-1062.4.3.el7.x86_64)
CPU=amd64 (2 logical CPUs) (0x1e845c000 RAM)
----------- Stack Backtrace -----------
(0x00007F9AA1AA99D2 [libj9prt29.so+0x3a9d2])
(0x00007F9AA1A8E283 [libj9prt29.so+0x1f283])
(0x00007F9AA1AA9A4E [libj9prt29.so+0x3aa4e])
(0x00007F9AA1AA9B44 [libj9prt29.so+0x3ab44])
(0x00007F9AA1A8E283 [libj9prt29.so+0x1f283])
(0x00007F9AA1AA98AB [libj9prt29.so+0x3a8ab])
(0x00007F9AA2192B44 [libj9vm29.so+0x9eb44])
(0x00007F9AA1A8E283 [libj9prt29.so+0x1f283])
(0x00007F9AA2192D98 [libj9vm29.so+0x9ed98])
(0x00007F9AA1A8D74A [libj9prt29.so+0x1e74a])
(0x00007F9AA3EC2890 [libpthread.so.0+0x12890])
---------------------------------------`

**Comment by JasonFengJ9** [comment]
System core files are required for investigation to determine the actual cause.
Can this Segmentation error be reproduced with latest OpenJ9 release [1]?

[1] https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/tag/jdk8u232-b09_openj9-0.17.0

**Comment by pshipton** [comment]
@roded pls open a new Issue for your crash, and include the diagnostic files produced. It doesn't appear to be the same as this Issue.

**Comment by roded** [comment]
@pshipton, we don't currently have the diagnostic files for the above crash. We'll try to get them if/when it reoccurs. Should I open another ticket only once we obtain the diagnostic files? Also, can you tell from the above exception if there's any relation to #8181?

Much obliged.

**Comment by pshipton** [comment]
@roded it does not look the same as #8181. We won't be able to investigate the problem without the diagnostic files, it makes sense to wait on opening a new Issue until diagnostic files are available.
